tests/test_rrt_star_reeds_shepp.py

```diff
@@ -39,9 +39,7 @@ def test3():
                                              max_iter=100, step_size=step_size)
     rrt_star_reeds_shepp.set_random_seed(seed=8)
     path = rrt_star_reeds_shepp.planning(animation=False)
-    for i in range(len(path)-1):
-        # + 0.00000000000001 for acceptable errors arising from the planning process
-        assert m.math.dist(path[i][0:2], path[i+1][0:2]) < step_size + 0.00000000000001
```

Comment:
Why this code should be changed?

Reply:
This specific test seems to be a bit off. I have explained this in detail in [issue #994](https://github.com/AtsushiSakai/PythonRobotics/issues/944).

Reply:
It looks like this test just check the path point interval length just less than `step_size`.
You mean this test starts failing after this PR changes?

Reply:
Even if the test passes with the original code, the paths generated are not appropriate as explained in the issue. However, with this PR's changes, the test also fails. 

Here is a possible explanation I have for this:

1. `calc_interpolate_dists_list` function in reeds_shepp_path_planning.py uses `interp_dists = np.arange(0.0, length, d_dist)` in line 328 to generate steps for a given path length. So if `length < step_length` then `interp_dists = [0]`. 

2. Older code has a path_type: straight_curve_straight, which is absent in the PR version.  Generally, straight maneuvers are greater then step length but curved maneuvers are not. So curved paths end up getting skipped in the path returned by reeds_shepp_path_planning. And what we see are just straight maneuvers appended to each other.

3. The newer code has does not have the path type and is supposed to return shorter paths. Thus, any path returned by newer code is node able to have a maneuver greater than step_size (20 units). However, the new code passes the test with `step_size = 0.2`.

Thus, either the step_size should be reduced appropriately or no path should be expected with the current step_size.

Reply:
OK. Make sense. I agree when the `step_size` is too big, the function should return None instead of invalid path points.
Did you add any validation for the too big `step_size`?
And, could you please change test function name from `test3` to `test_too_big_step_size` or something?

Reply:
I can add the validation if required. Will do the renaming.